Runtimes/Core/cmake/modules/gyb.cmake

```diff
@@ -0,0 +1,47 @@
+find_package(Python3 REQUIRED)
+
+# Create a target to expand a gyb source
+# target_name: Name of the target
+# FLAGS  list of flags passed to gyb
+# DEPENDS list of dependencies
+# COMMENT Custom comment
+function(gyb_expand source output)
+  set(flags)
+  set(arguments)
+  set(multival_arguments FLAGS DEPENDS)
+  cmake_parse_arguments(GYB "${flags}" "${arguments}" "${multival_arguments}" ${ARGN})
+
+  get_filename_component(full_output_path ${output} ABSOLUTE BASE_DIR "${CMAKE_CURRENT_BINARY_DIR}")
+  get_filename_component(dir "${full_output_path}" DIRECTORY)
+  get_filename_component(fname "${full_output_path}" NAME)
+
+  file(READ "${source}" gyb_src)
+  string(REGEX MATCHALL "\\\$\{[\r\n\t ]*gyb.expand\\\([\r\n\t ]*[\'\"]([^\'\"]*)[\'\"]" gyb_expand_matches "${gyb_src}")
+  foreach(match ${gyb_expand_matches})
+    string(REGEX MATCH "[\'\"]\([^\'\"]*\)[\'\"]" gyb_dep "${match}")
+    list(APPEND gyb_expand_deps "${CMAKE_MATCH_1}")
+  endforeach()
+  list(REMOVE_DUPLICATES gyb_expand_deps)
+
+  set(utils_dir "${PROJECT_SOURCE_DIR}/../../utils/")
```

Comment:
I am seeing this `${PROJECT_SOURCE_DIR}/../..` bit in several places. I don't think `Runtimes/Core` has a lot of chances of changing directory levels, but in case it does, chasing all these might not be fun. I was thinking in having a `SWIFT_TOP_LEVEL_SOURCE_DIR` calculated once after the `project()` call, which can be used in the rest of the files. It also happens to document how to get to the top level source dir, if necessary.